plugin/src/withGoogleSignIn.ts

```diff
@@ -0,0 +1,27 @@
+import {
+  ConfigPlugin,
+  AndroidConfig,
+  IOSConfig,
+  createRunOncePlugin,
+  withPlugins,
+} from '@expo/config-plugins';
+
+const pkg = require('@react-native-google-signin/google-signin/package.json');
+
+/**
+ * Apply google-signin configuration for Expo SDK 44 projects.
+ */
+const withGoogleSignIn: ConfigPlugin = (config) => {
+  return withPlugins(config, [
+    // Android
+    AndroidConfig.GoogleServices.withClassPath,
+    AndroidConfig.GoogleServices.withApplyPlugin,
+    AndroidConfig.GoogleServices.withGoogleServicesFile,
+
+    // iOS
+    IOSConfig.Google.withGoogle,
+    IOSConfig.Google.withGoogleServicesFile,
```

Comment:
This isn't required since all of the plugins are default but I could see the value in adding them regardless for potential backwards compatibility.

The main drawback being that the duplicate operations would increase prebuild time. This can be fixed internally by memoizing the plugins using history API.